### ISSUE TITLE
convert partial if name got space inside

### ISSUE THREAD

**Issue by SeuperHakkerJa** [jupyter/nbconvert#965]
When using the command line to convert the Ipython notebook, the conversion tends to fail. The symptom is either conversion failed directly or a part of the fail left undone making the converted file incomplete.

**Comment by SeuperHakkerJa** [comment]
```jupyter nbconvert --to pdf long\ name.ipynb ``` provide partially converted pdf

**Comment by MSeal** [comment]
I am on vacation but can look into this when I get back unless someone else
beats me to investigating.

**Comment by SeuperHakkerJa** [comment]
No problem! Thanks

**Comment by MSeal** [comment]
Uhg, ok reproduced. This is a weird issue and one of us will need to dig a bit to figure out why. I've got a few things on my plate but I'll work towards it when I get some more critical items fixed elsewhere if no one else gets to this.

**Comment by mpacer** [comment]
Could you provide a complete minimal working example? i'm having a little confusion around what exactly that partial failure is and when I try to do this on my notebooks with spaces in the names it works fine.

**Comment by MSeal** [comment]
Failed Outcome (`jupyter nbconvert --to pdf "highlight date.ipynb"`)
[highlight date.pdf](https://github.com/jupyter/nbconvert/files/2998550/highlight.date.pdf)
Successful Outcome (`jupyter nbconvert --to pdf highlight_dates_run_cli.ipynb`)
[highlight_dates_run_cli.pdf](https://github.com/jupyter/nbconvert/files/2998551/highlight_dates_run_cli.pdf)
Source notebook (with extension changed to .txt so github will load it):
[highlight_dates_run_cli.txt](https://github.com/jupyter/nbconvert/files/2998716/highlight_dates_run_cli.txt)

**Comment by mpacer** [comment]
Thank you! 

To be clear, this only applies to pdf conversion, correct?

From looking at this output: 

```
! Missing $ inserted.
<inserted text>
                $
l.384 ...light date_files/highlight date_12_1.png}

?
! Emergency stop.
<inserted text>
                $
l.384 ...light date_files/highlight date_12_1.png}
```

I'm pretty sure the solution to this will relate to how we create links to anchors in https://github.com/jupyter/nbconvert/blob/master/nbconvert/filters/filter_links.py#L33-L35. 

Specifically, we may need to do something like that when we're making the image include statements.

However, I think the template that will need to change is probably this part

https://github.com/jupyter/nbconvert/blob/12e148f1e8e2d4f07cd6b9df0a75b0454b7a5d55/nbconvert/templates/latex/document_contents.tplx#L42-L56

**Comment by SeuperHakkerJa** [comment]
not sure if it helps but i personally convert jupyter notebook into pdf using a template i wrote myself but the same situation did happene even with --template.

**Comment by SeuperHakkerJa** [comment]
not sure if it helps but i personally convert jupyter notebook into pdf using a template i wrote myself but the same situation did happene even with --template.

**Comment by llimeht** [comment]
Workarounds are:

  * rename the notebook file to not have a space
  * use `pdflatex` rather than `xelatex` for compiling the tex file (undoing the change made in https://github.com/jupyter/nbconvert/commit/1db4edc503269dd2e42ce07abb1b20aa35d18029 or by doing the conversion in two steps: `jupyter nbconvert --to latex 'foo bar.ipynb'; pdflatex 'foo bar'`)
  * set the output filename to be something that doesn't have a space (`jupyter nbconvert --to pdf --output foo 'foo bar.ipynb'` will make foo.pdf and uses `foo/` as the temp dir rather than `foo bar/`)

As a longer term fix, one of the following could work

  * Add lots of extra escaping to filenames for latex.
  * Omit all special characters from the temp filenames that nbconvert is making, ensuring there are no underscores, spaces or multiple dots in them. Trying to escape special characters tends to be the gift that keeps on giving, and it's easier to use safe filenames than to figure out how to escape them for each different latex engine.
  * Use a directory name that isn't going to freak latex out. Could the entire conversion chain work in a directory from `tempfile.TemporaryDirectory`, with the latex files and then pdf in a location with safe filenames, finally moving the pdf back to the desired output location?

(This seems closely related to #633, possibly a duplicate.)

**Comment by MSeal** [comment]
Based on reading the code and the sensitivity of the latex systems upstream I think the TemporaryDirectory pattern will be the most reliable solution here. I'll see if I can make a quick PR to do just that.

**Comment by t-makaro** [comment]
> Failed Outcome (jupyter nbconvert --to pdf "highlight date.ipynb")
highlight date.pdf
Successful Outcome (jupyter nbconvert --to pdf highlight_dates_run_cli.ipynb)
highlight_dates_run_cli.pdf
Source notebook (with extension changed to .txt so github will load it):
highlight_dates_run_cli.txt

I just tested this on master and both produce identical results.

I believe that this issue and #633 (which are duplicate issues) are fixed by #1053. That PR includes a fix for ```\usepackage{grffile}``` that makes spaces in files names work properly.
